components/web_discovery/browser/server_config_loader.cc

```diff
@@ -332,14 +379,10 @@ void ServerConfigLoader::OnPatternsFileLoaded(
     SchedulePatternsRequest();
     return;
   }
-  auto parsed_patterns = ParsePatterns(*patterns_json);
-  if (!parsed_patterns) {
-    local_state_->ClearPref(kPatternsRetrievalTime);
-    SchedulePatternsRequest();
-    return;
-  }
-  last_loaded_patterns_ = std::move(parsed_patterns);
-  patterns_callback_.Run();
+  background_task_runner_->PostTaskAndReplyWithResult(
+      FROM_HERE, base::BindOnce(&ParsePatterns, *patterns_json),
```

Comment:
why are you copying `patterns_json`? why can't you just parse it as a part of `ReadPatternsFile` in a single background thread call?